tests/robustness/makefile.mk

```diff
@@ -1,3 +1,7 @@
+.PHONY: test-robustness-reports
+test-robustness-reports:
+	cd ./tests && go test ./robustness/validate -v --run TestDataReports
```

Comment:
I think this depends on setup to setup, but we should add in a `toolchain` directive in the `go.mod` under `tests/`: https://github.com/etcd-io/etcd/blob/main/tests/go.mod

Reason for this is similar to: https://github.com/golang/go/issues/62278

And I had to add this directive to make it work since go will try to download the toolchain that is specified by the go directive of the `go.mod` in the directory you're running in. It will take the go directive if the toolchain is missing. So we probably need to do something similar to what was done in the `go.mod` at the root.

Reply:
Don't understand what you are proposing. If you want to propose to add `toolchain` directive, please file an issue. Don't think we should block this PR on this.

Reply:
What I mean is if I run the code here following the instructions this is what I get:
```
❯ make test-robustness-reports
cd ./tests && go test ./robustness/validate -v --run TestDataReports
go: downloading go1.22 (darwin/amd64)
go: download go1.22 for darwin/amd64: toolchain not available
make: *** [test-robustness-reports] Error 1
```
and the above linked issue explains why.

Created a PR here @serathius: https://github.com/etcd-io/etcd/pull/17734

Reply:
Thank you, I didn't know that the command could not work on some golang setups.

Reply:
Would an alternative be to do something like:

```diff
test-robustness-reports:
-	cd ./tests && go test ./robustness/validate -v --run TestDataReports
+	cd ./tests && GOTOOLCHAIN=$(cat ../.go-version) go test ./robustness/validate -v --run TestDataReports
```

We put a lot of effort removing a bunch of duplicated version references in the main repo so we only had one main file to update when we bump go. Keen to try and keep that simplicityif possible.